deduplication/src/main/java/gov/cdc/nbs/deduplication/data_elements/DataElementsService.java

```diff
@@ -0,0 +1,161 @@
+
+            recordLinkageClient.put()
+                    .uri("/algorithm/dibbs-enhanced")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .body(updatedConfigJson)
+                    .retrieve()
+                    .body(Void.class);
+        } catch (JsonProcessingException e) {
+            throw new DataElementConfigurationException("Error serializing algorithm configuration to JSON", e);
+        } catch (Exception e) {
+            throw new DataElementConfigurationException("Failed to update algorithm configuration", e);
+        }
+    }
+
+
+    public void updateKwargs(JsonNode kwargs, Map<String, DataElementsDTO.DataElementConfig> dataElements) {
+        if (!(kwargs instanceof ObjectNode)) {
+            throw new IllegalArgumentException("kwargs must be an ObjectNode");
+        }
+        ObjectNode kwargsObject = (ObjectNode) kwargs;
+
+        // Get or create the thresholds and log_odds nodes
+        ObjectNode thresholdsNode = (ObjectNode) kwargsObject.get("thresholds");
+        if (thresholdsNode == null) {
+            thresholdsNode = kwargsObject.putObject("thresholds"); // Ensure it's created
+        }
+
+        ObjectNode logOddsNode = (ObjectNode) kwargsObject.get("log_odds");
+        if (logOddsNode == null) {
+            logOddsNode = kwargsObject.putObject("log_odds"); // Ensure it's created
+        }
+
+        // Map DataElementsDTO fields to API field names
+        Map<String, String> fieldMapping = Map.ofEntries(
+                Map.entry("firstName", "FIRST_NAME"),
+                Map.entry("lastName", "LAST_NAME"),
+                Map.entry("dateOfBirth", "BIRTHDATE"),
+                Map.entry("currentSex", "SEX"),
+                Map.entry("race", "RACE"),
+                Map.entry("suffix", "SUFFIX"),
+                Map.entry("streetAddress1", "ADDRESS"),
+                Map.entry("city", "CITY"),
+                Map.entry("state", "STATE"),
+                Map.entry("zip", "ZIPCODE"),
+                Map.entry("county", "COUNTY"),
+                Map.entry("telephone", "PHONE"),
+                Map.entry("telecom", "PHONE"),
+                Map.entry("email", "EMAIL"),
+                Map.entry("accountNumber", "AN"),
+                Map.entry("driversLicenseNumber", "DL"),
+                Map.entry("medicaidNumber", "MA"),
+                Map.entry("medicalRecordNumber", "MR"),
+                Map.entry("medicareNumber", "MC"),
+                Map.entry("nationalUniqueIdentifier", "NI"),
+                Map.entry("patientExternalIdentifier", "PI"),
+                Map.entry("patientInternalIdentifier", "PT"),
+                Map.entry("personNumber", "PN"),
+                Map.entry("socialSecurity", "SS"),
+                Map.entry("visaPassport", "VS"),
+                Map.entry("wicIdentifier", "WC")
+        );
```

Comment:
Suggestion: move this to the class level and mark it as final. Declaring it within the method will re-create it every time this method is called.